<launch>

  <arg name="robot" default="orcvio_robot"/>
  <arg name="fixed_frame_id" default="world"/>

  <!-- bag parameters -->
  <arg name="path_bag" default="/media/erl/disk1/orcvio/opcity_smooth/docker_compose_opcity_20cars_quad.bag" />
  <!-- <arg name="path_bag" default="/media/erl/disk1/orcvio/opcity_smooth/docker_compose_opcity_20cars_quad_short.bag" /> -->
  <arg name="visualize_rviz" default="true" />

  <!-- System Manager Nodelet  -->
  <group ns="$(arg robot)">
    <node pkg="nodelet" type="nodelet" name="system"
      args='standalone orcvio/SystemNodelet'
      output="screen">

      <param name="fixed_frame_id" value="$(arg fixed_frame_id)"/>
      <param name="child_frame_id" value="odom"/>

      <remap from="~imu" to="/falcon/IMU"/>
      <remap from="~cam0_image" to="/falcon/cam_left/image_raw"/>

      <param name="config_file_template" value="$(find orcvio)/../../../config/unity.yaml"/>
      <param name="config_file" value="$(find orcvio)/../../../cache/unity_exp.yaml"/>
      <param name="result_file" value="$(find orcvio)/../../../cache/temp_rmse.txt"/>

      <param name="load_groundtruth_flag" type="int" value="0"/>
      <!-- <param name="use_left_perturbation_flag" type="int" value="1" /> -->

      <param name="output_dir_traj"  type="string" value="/home/erl/Workspace/orcvio_cpp/cache" />

    </node>
  </group>

  <node name="publish_gt_path" pkg="orcvio" type="publish_gt_path" output="screen" clear_params="true"
        required="false">
        
      <!-- for quadrotor -->
      <param name="topic_pose"  type="string" value="/unity_command/ground_truth/falcon/pose" />
      <!-- for car  -->
      <!-- <param name="topic_pose"  type="string" value="/unity_command/ground_truth/husky/pose" /> -->

      <param name="output_dir_traj"  type="string" value="/home/erl/Workspace/orcvio_cpp/cache" />  

  </node>

    <!-- play the dataset -->
    <!-- clock start 20  -->
    <node pkg="rosbag" type="play" name="rosbag" args="--clock --start 20 $(arg path_bag)" required="true"/>
    
    <!-- open rviz  -->
    <group if="$(arg visualize_rviz)" >
      <node type="rviz" name="rviz" pkg="rviz" args="-d $(find orcvio)/../../../ros_wrapper/src/orcvio/launch/rviz/orcvio_unity.rviz" />
    </group>

</launch>